pkg/crc/network/nameservers.go

```diff
@@ -34,13 +36,48 @@ func GetResolvValuesFromInstance(sshRunner *ssh.Runner) (*ResolvFileValues, erro
 func CreateResolvFileOnInstance(sshRunner *ssh.Runner, resolvFileValues ResolvFileValues) error {
 	resolvFile, _ := CreateResolvFile(resolvFileValues)
 
+	sd := systemd.NewInstanceSystemdCommander(sshRunner)
+	// Check if ovs-configuration.service exist and if not then it is old bundle and use the same way to
+	// update resolve.conf file
+	if state, err := sd.Status("ovs-configuration.service"); err != nil || state == states.NotFound {
+		if err := replaceResolvConfFile(sshRunner, resolvFile); err != nil {
+			return fmt.Errorf("error updating resolve.conf file: %s", err)
+		}
+		return nil
+	}
+
+	if err := sd.Start("ovs-configuration.service"); err != nil {
+		return err
+	}
+
+	nameservers := strings.Join(resolvFileValues.GetNameServer(), ",")
+	searchDomains := strings.Join(resolvFileValues.GetSearchDomains(), ",")
+
+	if err := updateNetworkManagerConfig(sshRunner, nameservers, searchDomains); err != nil {
+		return fmt.Errorf("failed to update NetworkManager configuration: %s", err)
+	}
+
+	return sd.Restart("NetworkManager.service")
+}
+
+func replaceResolvConfFile(sshRunner *ssh.Runner, resolvFile string) error {
 	err := sshRunner.CopyDataPrivileged([]byte(resolvFile), "/etc/resolv.conf", 0644)
 	if err != nil {
 		return fmt.Errorf("Error creating /etc/resolv on instance: %s", err.Error())
 	}
 	return nil
 }
 
+func updateNetworkManagerConfig(sshRunner *ssh.Runner, nameservers, searchDomains string) error {
+	// When ovs-configuration service is running we are sure that name of the connection is ovs-if-br-ex
+	_, stderr, err := sshRunner.RunPrivileged("Update resolve.conf file", "nmcli", "con", "modify", "ovs-if-br-ex",
```

Comment:
`resolve.conf` - typo

Is there a way of programmatically getting the name of the connection to use instead of hardcoding it?

Reply:
> Is there a way of programmatically getting the name of the connection to use instead of hardcoding it?

I wish, I tried my search foo but didn't find anything. Let me know if you have some pointer, that would be great.

Reply:
Might be worth a `nmcli conn show --active ovs-if-br-ex` call first to ensure it exists?

Reply:
if it not exist then what would be our fallback? because if it not exist then if will anyway fails when adding the domain and nameserver .

Reply:
We could output a good error message instead of something possibly hard to understand.